backend/test/integration/job_api_test.go

```diff
@@ -288,6 +288,24 @@ func (s *JobApiTestSuite) TestJobApis() {
 	assert.Equal(t, 1, totalSize)
 	assert.Equal(t, "hello world", jobs[0].Name)
 
+	/* ---------- List the jobs, filtered by created_at, only return the previous two jobs ---------- */
+	filterTime := time.Now().Unix()
+	_, err = s.jobClient.Create(createJobRequest)
+	// Check total number of jobs to be 3
+	jobs, totalSize, _, err = test.ListAllJobs(s.jobClient, s.resourceNamespace)
+	assert.Nil(t, err)
+	assert.Equal(t, 3, totalSize)
+	assert.Equal(t, 3, len(jobs))
+	// Check number of filtered jobs finished before filterTime to be 2
+	jobs, totalSize, _, err = test.ListJobs(
+		s.jobClient,
+		&jobparams.ListJobsParams{
+			Filter: util.StringPointer(`predicates { key: "created_at" op: LESS_THAN_EQUALS long_value: ` + fmt.Sprint(filterTime) + ` }`)},
+		s.resourceNamespace)
+	assert.Nil(t, err)
+	assert.Equal(t, 2, len(jobs))
+	assert.Equal(t, 2, totalSize)
+
```

Comment:
It seems that we also need to remove the created jobs and runs as they may ruin other tests. Another alternative is to (1) get existing jobs/runs, (2) compute the timestamp in the middle, (3) list them again with filtering at the middle timestamp, (4) assert that we get expected number of jobs/runs after filtering.
```
integration-test-gmvs5-1366116216:         	Test:       	TestJobApi/TestJobApis
integration-test-gmvs5-1366116216:     job_api_test.go:349: 
integration-test-gmvs5-1366116216:         	Error Trace:	job_api_test.go:349
integration-test-gmvs5-1366116216:         	Error:      	Expected nil, but got: &errors.errorString{s:"expected runs to be length 1, got: 2"}
```

Reply:
Thank you for finding the error. It turned out to be an invalid filter string, which I just fixed. The other test failed due to the set wait time exceeded the time out limit. Since I also don't think it's a good idea to make our e2e test even longer, I opted to use `created_at` for filter. Currently the integration test is still a bit flaky, but I'm not sure why.

Reply:
Added some small delay to fix the flakiness.